src/library/process_vehicles.py

```diff
@@ -11,13 +11,25 @@ def process_vehicles(ctx, datetime):
     agency = ctx.config["METRO_AGENCY"]
     lines = ctx.config["METRO_LINES"]
     date = datetime.in_tz(ctx.config["TIMEZONE"]).format("YYYY-MM-DD")
+    preprocessed_path = ctx.tmp.get_abs_path(f"tracking/{agency}/preprocessed.csv")
+    processed_path = ctx.tmp.get_abs_path(f"tracking/{agency}/processed.csv")
+
+    allData = pd.read_csv(preprocessed_path, index_col=0)
+
+    try:
+        df = pd.read_csv(processed_path, index_col=0)
+    except:
+        df = pd.DataFrame(columns=["datetime", "trip_id", "direction_id", "relative_position"])
+
     for line in lines:
         try:
-            path = ctx.tmp.get_abs_path(f"tracking/{agency}/{line}/preprocessed.csv")
-            df = pd.read_csv(path, index_col=0)
+            #path = ctx.tmp.get_abs_path(f"tracking/{agency}/{line}/preprocessed.csv")
+            lineData = allData[allData['line']==line] 
             tracks = get_track(line, ctx.config["LOCAL_DATA"])
-            processed = process_raw_vehicles(df, tracks)
-            ctx.tmp.write(f"tracking/{agency}/{line}/processed.csv", processed.to_csv())
-        except:
+            processed = process_raw_vehicles(lineData, tracks)
+            df = pd.concat([df, processed], ignore_index=True, sort=False, join="inner")
```

Comment:
Unfortunately there are a couple of problems with this:

1. `processed` data is not guaranteed to be new data only unless we wipe the preprocessed file every time at the _end_ of **this** script. Currently our cron job only wipes it once a day. I'm not opposed to changing that though.

2. The bigger problem is that `process_raw_vehicles` (which is too opaque right now, we might move some of its functionality back into this script) does something that requires its input to contain an entire day's worth of data. The [getTrips](https://github.com/metro-ontime/performance_tracker/blob/47092ed454ec92f674c227adf9a2d59d38d94012/src/library/analyzer/process_vehicles.py#L77) function goes through the vehicle observations in datetime order and numbers the trips as it sees vehicles switch direction. So unless we can somehow "pick up where it left off" then we need to reprocess the entire day of data here.

If we stick with just processing an entire day of data every time we run this script and overwrite the output file each time then there's no need to read anything in other than the `preprocessed` csv. It might be easier to stick with this behavior for now and consider optimizing later - once we have it working properly.

Reply:
I agree that its easier to move ahead with processing an entire day of data for now. I've rewritten it accordingly and we can revisit this at a later time if we feel the need.